start.py

```diff
@@ -265,7 +283,10 @@ async def main(args: argparse.Namespace, pwd: str) -> None:
     initial_start_commands = [
         f"cd {pwd}",
         "source install/setup.bash",
+        "export ROS_DOMAIN_ID=42" if args.sim == "real" else "",
```

Comment:
Hmm, this goes against lines 288-289.

What if instead we change the arg to `real_domain_id` with a default of 42, and then that domain ID is only used if args.sim == "real"?

Also, in terms of how to write the conditional, I prefer the construction on 288-289 compared to this, because this code unnecessarily adds an empty string in the else case, whereas the construction on 288-289 does not.